codegen/smithy-aws-typescript-codegen/build.gradle.kts

```diff
@@ -13,14 +13,44 @@
  * permissions and limitations under the License.
  */
 
+import software.amazon.smithy.model.node.Node
+
 description = "Generates TypeScript code for AWS protocols from Smithy models"
 extra["displayName"] = "Smithy :: AWS :: Typescript :: Codegen"
 extra["moduleName"] = "software.amazon.smithy.aws.typescript.codegen"
 
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath("software.amazon.smithy:smithy-model:[1.7.0, 1.8.0[")
+    }
+}
+
 dependencies {
     api("software.amazon.smithy:smithy-aws-cloudformation-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-aws-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-waiters:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-aws-iam-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy.typescript:smithy-typescript-codegen:0.3.0")
 }
+
+tasks.register("set-aws-sdk-version") {
+    doLast {
+        var packageJson = Node.parse(project.file("../../package.json").readText()).expectObjectNode()
```

Comment:
The version for individual packages may differ and needs to be read separately.

For example, the package `@aws-sdk/middleware-logger` did not have versions `3.16.0` and `3.17.0` https://github.com/aws/aws-sdk-js-v3/blob/main/packages/middleware-logger/CHANGELOG.md

This PR needs to be updated to read version of individual dependencies.

Reply:
I now walk the top level of both `clients/` and `/packages`, pull the name and version out of every package.json encountered, and bake them into the codegen JAR for reference at runtime.